python/lsst/daf/butler/_butler.py

```diff
@@ -333,10 +333,13 @@ def makeRepo(root: str, config: Union[Config, str, None] = None, standalone: boo
             if not os.path.isdir(uri.ospath):
                 safeMakeDir(uri.ospath)
         elif uri.scheme == "s3":
-            s3 = boto3.resource("s3")
-            # implies bucket exists, if not another level of checks
-            bucket = s3.Bucket(uri.netloc)
-            bucket.put_object(Bucket=uri.netloc, Key=uri.relativeToPathRoot)
+            # implies bucket already exists
+            s3 = boto3.client("s3")
+            if uri.relativeToPathRoot != ".":
+                # to be dir-equivalent, S3 this requires URIs ending on '/'
+                if not uri.geturl().endswith('/'):
+                    uri.updateFile(uri.relativeToPathRoot+'/')
```

Comment:
If ButlerURI had a constructor flag along the lines of `forceDirectory=True` (defaulting to false) then maybe we wouldn't need these hacks.